crates/forge/tests/it/zk/cheats.rs

```diff
@@ -210,6 +214,184 @@ forgetest_async!(test_zk_use_factory_dep, |prj, cmd| {
+
+    contract Counter {
+        uint256 public count;
+        function increment() external {
+            count++;
+        }
+    }
+    "#,
+    )
+    .unwrap();
+
+    let node = ZkSyncNode::start().await;
+    let (_, private_key) = ZkSyncNode::rich_wallets()
+        .next()
+        .map(|(addr, pk, _)| (addr, pk))
+        .expect("No rich wallets available");
+
+    let _ = cmd
+        .args([
+            "create",
+            "src/Counter.sol:Counter",
+            "--zksync",
+            "--private-key",
+            private_key,
+            "--rpc-url",
+            node.url().as_str(),
+        ])
+        .assert_success()
+        .get_output()
+        .stdout_lossy();
+
+    cmd.forge_fuse();
+
+    prj.add_script(
+        "SimpleScript.s.sol",
+        r#"
+import "forge-std/Script.sol";
+import {Counter} from "../src/Counter.sol";
+contract SimpleScript is Script {
+    function run() external {
+        // zk raw transaction
+        vm.startBroadcast();
+        // This raw transaction comes from cast mktx of increment() to Counter contract
+        // `cast mktx "0x9086C95769C51E15D6a77672251Cf13Ce7ebf3AE" "increment()" --rpc-url http://127.0.0.1:49204 --private-key "0x3d3cbc973389cb26f657686445bcc75662b415b656078503592ac8c1abb8810e" --zksync --nonce 1`
+        vm.broadcastRawTransaction(
+            hex"71f88501808402b275d08304d718949086c95769c51e15d6a77672251cf13ce7ebf3ae8084d09de08a01a00d798053cc7a75a78d49adc0b893d9ad91f5301bb264eb2848979859fc366dc7a06469714a3ec85003c3f52ff668299c3c01e0a43be5ec0529cc204fd53be1629e82010494bc989fde9e54cad2ab4392af6df60f04873a033a80c08080"
+        );
+        vm.stopBroadcast();
+    }
+}
+"#,
+    )
+    .unwrap();
+
+    cmd.args([
+        "script",
+        "--zksync",
+        "--private-key",
+        private_key,
+        "--rpc-url",
+        node.url().as_str(),
+        "--broadcast",
+        "--slow",
+        "--non-interactive",
+        "SimpleScript",
+    ])
+    .assert_success()
+    .get_output()
+    .stdout_lossy();
+
+    let run_latest = foundry_common::fs::json_files(prj.root().join("broadcast").as_path())
+        .find(|file| file.ends_with("run-latest.json"))
+        .expect("No broadcast artifacts");
+
+    let content = foundry_common::fs::read_to_string(run_latest).unwrap();
+
+    let json: serde_json::Value = serde_json::from_str(&content).unwrap();
+    let txns = json["transactions"].as_array().expect("broadcastable txs");
+
+    // check that the txs have the correct function and contract address
+    assert_eq!(txns.len(), 1);
+    txns[0]["function"].as_str().expect("function name").contains("increment");
+    txns[0]["contractAddress"]
+        .as_str()
+        .expect("contract address")
+        .contains("0x9086c95769c51e15d6a77672251cf13ce7ebf3ae");
+
+    // check that the txs have strictly monotonically increasing nonces
+    assert!(txns.iter().filter_map(|tx| tx["nonce"].as_u64()).is_sorted_by(|a, b| a + 1 == *b));
```

Comment:
this was specifically for the factory deps batching test because we were missing nonces, but here you also only have 1 tx so no need for it